integration_test/test_videoroom/test/integration/containerised_test.exs

```diff
@@ -0,0 +1,203 @@
+  # How many result entries will be discarded when checking the ratio between frames
+  # received from regular browsers and frames received from the one with packet loss applied
+  @packet_loss_results_ratio_warmup 5
+  # Max allowed difference between received frames from separate browsers, at each stat entry
+  @max_frame_count_difference 10
+
+  @tag timeout: @max_test_duration
+  @tag :packet_loss_test
+  test "packet loss on one browser" do
+    TestVideoroom.Integration.ResultReceiver.start_link(browser_count: 3, parent: self())
+
+    results = receive_loop()
+
+    for {tag, tag_results} <- prepare_results(results) do
+      assert_packet_loss_stats(tag, tag_results)
+    end
+  end
+
+  defp receive_loop() do
+    receive do
+      {:results, results} ->
+        results
+
+      {:enable_packet_loss, name} ->
+        Logger.info("#{inspect(name)} reports ready to enable packet loss")
+        File.write!(Path.join(@shared_folder, "ENABLE_PACKET_LOSS"), <<>>)
+        receive_loop()
+
+      other ->
+        raise("Received unexpected message: #{inspect(other)}")
+    end
+  end
+
+  # We need to transform the results to be able to work with them easier
+  # After calling `prepare_results/1`, the results will have the following structure:
+  #   %{
+  #     tag0: [
+  #       %{                        # Entries from all browsers, from around the same time
+  #         "browser0" => %{          # Browser-receiver (the one we read stats from) =>
+  #           "browser1" => %{...},     # Browser-sender => single stat entry
+  #           "browser2" => %{...}      # Browser-sender => single stat entry
+  #         },
+  #         "browser1" => %{          # as above
+  #           "browser0" => %{...},
+  #           "browser2" => %{...}
+  #         }
+  #       },
+  #       %{                        # Entries from all browsers (another reading)
+  #         ...
+  #       },
+  #       ... # (as many entries as there have been readings for the given tag)
+  #     ],
+  #     tag1: [
+  #       ...
+  #     ]
+  #   }
+  defp prepare_results(results) do
+    peer_id_to_browser_id = get_peer_id_mapping(results)
+
+    Map.new(@packet_loss_tags_to_check, fn tag ->
+      acc_list =
+        Enum.reduce(results, [], fn {browser_id, browser_results}, acc_list ->
+          tag_results = browser_results[tag][browser_id]
+
+          acc_list =
+            if Enum.empty?(acc_list) do
+              for _entry <- tag_results, do: %{}
+            else
+              acc_list
+            end
+
+          Enum.zip_with(acc_list, tag_results, fn acc_single, res ->
+            Map.put(acc_single, browser_id, label_stat_entries(res, peer_id_to_browser_id))
+          end)
+        end)
+
+      {tag, acc_list}
+    end)
+  end
+
+  defp get_peer_id_mapping(results) do
+    Map.new(results, fn {browser_id, browser_results} ->
+      peer_id =
+        browser_results[@sender_stats_tag][browser_id]
+        |> hd()
+        |> Map.get("peerId")
```

Comment:
As we are using this `"peerId"` here and below, I would move this to a constant.